NitroxClient/MonoBehaviours/NitroxDebugManager.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections.Generic;
+using NitroxClient.Debuggers;
+using UnityEngine;
+
+namespace NitroxClient.MonoBehaviours
+{
+    public class NitroxDebugManager : MonoBehaviour
+    {
+        public readonly KeyCode EnableDebuggerHotkey = KeyCode.F7;
+        public readonly List<BaseDebugger> Debuggers;
+        private readonly HashSet<BaseDebugger> prevActiveDebuggers = new HashSet<BaseDebugger>();
+        private bool isDebugging;
+        private Rect windowRect;
+
+        private NitroxDebugManager()
+        {
+            Debuggers = new List<BaseDebugger>
+            {
+                new SceneDebugger(),
+                new NetworkDebugger()
+            };
+        }
+
+        public void OnGUI()
+        {
+            if (!isDebugging)
+            {
+                return;
+            }
+
+            // Main window to display all available debuggers.
+            windowRect = GUILayout.Window(GUIUtility.GetControlID(FocusType.Keyboard), windowRect, DoWindow, "Nitrox debugging");
+
+            // Render debugger windows if they are enabled.
+            foreach (BaseDebugger debugger in Debuggers)
+            {
+                debugger.OnGUI();
+            }
+        }
+
+        public void Update()
+        {
+            if (Input.GetKeyDown(EnableDebuggerHotkey))
+            {
+                ToggleDebugging();
+            }
+
+            if (Input.GetKeyDown(KeyCode.C) && Input.GetKey(KeyCode.LeftControl))
```

Comment:
Do we want to check the rest of this stuff if we are not debugging?

Reply:
Only the ToggleCursor part worked when you're not debugging but it probably clutters the game more than it is useful (you can always just enable debugging with F7 then CTRL+C to toggle the cursor). I've changed it.